catchup/catchpointService.go

```diff
@@ -335,12 +339,27 @@ func (cs *CatchpointCatchupService) processStageLedgerDownload() (err error) {
 	return nil
 }
 
-// updateVerifiedAccounts update the user's statistics for the given verified accounts
-func (cs *CatchpointCatchupService) updateVerifiedAccounts(addedTrieHashes uint64) {
+// updateVerifiedCounts update the user's statistics for the given verified hashes
+func (cs *CatchpointCatchupService) updateVerifiedCounts(hashes [][]byte) {
 	cs.statsMu.Lock()
 	defer cs.statsMu.Unlock()
+
+	addedTrieAccountHashes := uint64(0)
+	addedTrieKVHashes := uint64(0)
+	for _, hash := range hashes {
+		if (hash[4] == 3) { // KV
+			addedTrieKVHashes++
+		} else {
+			addedTrieAccountHashes++
+		}
+	}
+
 	if cs.stats.TotalAccountHashes > 0 {
-		cs.stats.VerifiedAccounts = cs.stats.TotalAccounts * addedTrieHashes / cs.stats.TotalAccountHashes
+		cs.stats.VerifiedAccounts = cs.stats.TotalAccounts * addedTrieAccountHashes / cs.stats.TotalAccountHashes
+	}
+
+	if cs.stats.TotalKVs > 0 { // TODO Is TotalKVHashes needed?
```

Comment:
I do not think it is. I think it was only needed because of the way we had to estimate accounts due to the combined hashes.

An alternative is to _also_ remove that estimation mechanism, and use the same hash type check to report the actual number of _account_ hashes.

Reply:
Resolving - I've removed the TODO.